lib/xlibc-rt/check_stack.c

```diff
@@ -0,0 +1,25 @@
+#include <debug.h>
+#include "xboxkrnl/xboxkrnl.h"
+
+#ifdef DEBUG_CONSOLE
+    #define _print DbgPrint
+#else
+    #define _print debugPrint
+#endif
+
+void _xlibc_check_stack (unsigned int eax, unsigned int esp)
+{
+    // eax is the amount of requested memory
+    // esp is the stack before the call to __chkstk
```

Comment:
I'd suggest renaming the function parameters, especially because `eax=$eax`, but `esp=$ecx`.
Also this function itself has little to do with x86 anymore: it really only checks for stack bounds. So I'd rename `esp` to `stack-pointer`; and rename `eax` to `requested_size`.

*(Under the assumption this function was not written to be more similar to an existing implementation elsewhere; if so, I'd suggest to link to it in a comment)*

Reply:
I agree, that's a better way to name things.